File: app/src/main/java/com.example.safehome/System/ISystem.java
package com.example.safehome.System;

/**
 * Represents the Security system
 */
public interface ISystem {

	/**
	 * initializes some screen or object to enable you to program the system. This could be an admin function.
	 */
	void program();

	/**
	 * displays the system information
	 */
	void display();

	/**
	 * resets the system and returns true upon completion.
	 */
	boolean reset();

	/**
	 * 
	 * @param queryString
	 */
	String query(String queryString);

	/**
	 * arm the system
	 */
	boolean arm();

	/**
	 * disarm the system
	 */
	boolean disarm();

	/**
	 * Arm the system
	 * @return
	 */
	boolean isArmed();

}

File: app/src/main/kotlin/com.example.safehome/System/ISystem.kt
package com.example.safehome.System

/**
 * Represents the Security system
 */
interface ISystem {
    /**
     * initializes some screen or object to enable you to program the system. This could be an admin function.
     */
    fun program()

    /**
     * displays the system information
     */
    fun display()

    /**
     * resets the system and returns true upon completion.
     */
    fun reset(): Boolean

    /**
     *
     * @param queryString
     */
    fun query(queryString: String?): String?

    /**
     * arm the system
     */
    fun arm(): Boolean

    /**
     * disarm the system
     */
    fun disarm(): Boolean

    /**
     * Arm the system
     * @return
     */
    val isArmed: Boolean
}